pkg/cli/cli.go

```diff
@@ -983,23 +1001,165 @@ func printConnection(c *command, cvd RemoteCVDLocator, status ConnStatus) {
 	c.Printf("%s/%s: %s\n", cvd.Host, cvd.WebRTCDeviceID, state)
 }
 
-func buildAgentCmdArgs(flags *ConnectFlags, device string) []string {
-	args := []string{
-		ConnectionAgentCommandName,
-		device,
-	}
+func buildAgentCmdArgs(flags *ConnectFlags, device string, connectAgent string) []string {
+	args := []string{connectAgent, device}
 	return append(args, flags.AsArgs()...)
 }
 
-// Handler for the agent command. This is not meant to be called by the user
-// directly, but instead is started by the open command.
+func enableAdbConnection(socketPath string, enable bool) error {
+	var subcommand string
+	if enable {
+		subcommand = "connect"
+	} else {
+		subcommand = "disconnect"
+	}
+	adbconnectionArgs := []string{subcommand, "localfilesystem:" + socketPath}
+	adbconnectionCmd := exec.Command("adb", adbconnectionArgs...)
+	return adbconnectionCmd.Run()
+}
+
+// Handler for the webrtc agent command. This is not meant to be called by the
+// user directly, but instead is started by the open command.
+// The process runs as a background process of runConnectCommand, for proxying
+// connection to the ADB port located remotely.
+func runConnectionProxyAgentCommand(flags *ConnectFlags, c *command, args []string, opts *subCommandOpts) error {
+	if len(args) > 1 {
+		return fmt.Errorf("connection agent only supports a single device, received: %v", args)
+	}
+	if len(args) == 0 {
+		return errors.New("missing device")
+	}
+	device := args[0]
+	service, err := opts.ServiceBuilder(flags.CVDRemoteFlags, c.Command)
+	if err != nil {
+		return err
+	}
+
+	// Retrieving host to get IP address of host
+	host, err := findHost(service, flags.host)
+	if err != nil {
+		return fmt.Errorf("failed to find host")
+	}
+	if host.Docker == nil {
+		return errors.New("instance type should be Docker")
+	}
+
+	// Retriving ADB port of Cuttlefish device
+	controlDir := opts.InitialConfig.ConnectionControlDirExpanded()
+	cvd, err := findCVD(service, controlDir, flags.host, device)
+	if err != nil {
+		return fmt.Errorf("failed to find cvd: %w", err)
+	}
+	_, port, err := net.SplitHostPort(cvd.ADBSerial)
+	if err != nil {
+		return fmt.Errorf("failed to parse port from ADB serial: %w", err)
+	}
+	if port == "" {
+		return errors.New("failed to find port")
+	}
+
```

Comment:
how about extracting this block to a separate method like findDeviceAddress or something similar?

Reply:
Done.